src/traces/scattergl/calc.js

```diff
@@ -0,0 +1,38 @@
+/**
+* Copyright 2012-2017, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var isNumeric = require('fast-isnumeric');
+
+var Axes = require('../../plots/cartesian/axes');
+
+var calcColorscale = require('../scatter/colorscale_calc');
+var arraysToCalcdata = require('../scatter/arrays_to_calcdata');
+
+
+module.exports = function calc(gd, trace) {
+    var xa = Axes.getFromId(gd, trace.xaxis || 'x'),
+        ya = Axes.getFromId(gd, trace.yaxis || 'y');
+
+    var x = xa.makeCalcdata(trace, 'x'),
+        y = ya.makeCalcdata(trace, 'y');
+
+    var serieslen = Math.min(x.length, y.length), i;
+
+    // create the "calculated data" to plot
+    var cd = new Array(serieslen);
+    for(i = 0; i < serieslen; i++) {
+        cd[i] = {x: x[i], y: y[i]};
+    }
```

Comment:
@alexcjohnson there seems to be some intermediary or consumer relying on `cd` of that form, selection just does not work properly that way
```js
var x = cd[0].x, y = cd[0].y
var serieslen = Math.min(x.length, y.length), i;
for (i = 0; i < serieslen; i++) {
    var px = xa.c2p(x[i]), py = xa.c2p(y[i])
    if(polygon.contains([px, py])) {
        selection.push({
            x: px,
            y: py,
        });
    }
}
```
![image](https://user-images.githubusercontent.com/300067/26843055-efad6c24-4abc-11e7-80a6-6de09e2071f9.png)